### PR TITLE
Fixes unfathomable curios not blocking when equipped. Fixes the shielded component still blocking despite not having a 'wearer'.

### LINKED ISSUE #82068: Unfathomable Curio's shield doesn't work.
## Issue Summary

The shield doesn't block anything and the visual effect that screams "KICK ME, I'M A HERETIC" doesn't go away on hit.

## Round ID:

N/A

## Testmerges:

N/A

## Reproduction:

Wear the Unfathomable Curio as a Heretic, Get hit with a melee attack or shot, The shield doesn't block anything and doesn't dissipate.

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

The unfathomable curio now only blocks when equipped to the belt slot, and not when equipped in the hand.

The shielded component now no longer blocks despite not having a wearer, and does not block if the wearer is not the same as the owner in the ``hit_reaction()`` proc.

Fixes https://github.com/tgstation/tgstation/issues/82068

## Why It's Good For The Game

The curio was only blocking while in-hand, rather than how it should be; in the belt slot. Now it does what it is supposed to.

On top of that, this fixes what I think has been a bug for quite some time with shield_inhand just not being respected whatsoever with regards to whether or not the component worked. It only really determined whether or not you got the sprites added/removed, but didn't factor into the blocking ability whatsoever.

## Changelog

:cl:
fix: Unfathomable Curios now properly block as expected.
fix: The shielded component actually respects the shield_inhand when determining blocking potential.
fix: Ensures that the shielded component has a wearer before attempting to block, and that this wearer is the same as the owner of the item.
/:cl:

### CODE REVIEW

File: code/datums/components/shielded.dm [hunk 1]
```diff
@@ -155,6 +155,14 @@
 
 	COOLDOWN_START(src, recently_hit_cd, recharge_start_delay)
 
+	//No wearer? No block.
+	if(!wearer)
```
Comment (san7890) [inline code/datums/components/shielded.dm:159]:
```suggestion
	if(isnull(wearer))
```

typed variable

File: code/modules/antagonists/heretic/items/unfathomable_curio.dm [hunk 2]
```diff
@@ -44,9 +44,29 @@
...
+	UnregisterSignal(user, COMSIG_LIVING_CHECK_BLOCK)
+
+// Here we make sure our curio is only able to block while worn on the belt slot
+/obj/item/storage/belt/unfathomable_curio/proc/shield_reaction(mob/living/carbon/human/owner,
+	atom/movable/hitby,
+	damage = 0,
+	attack_text = "the attack",
+	attack_type = MELEE_ATTACK,
+	armour_penetration = 0,
+	damage_type = BRUTE
```
Comment (san7890) [inline code/modules/antagonists/heretic/items/unfathomable_curio.dm:63]:
```suggestion
	damage_type = BRUTE,
```

File: code/modules/antagonists/heretic/items/unfathomable_curio.dm [hunk 3]
```diff
@@ -44,9 +44,29 @@
...
+)
+	SIGNAL_HANDLER
+
+	if(SEND_SIGNAL(src, COMSIG_ITEM_HIT_REACT, owner, hitby, attack_text, 0, damage, attack_type) & COMPONENT_HIT_REACTION_BLOCK && owner.belt == src)
```
Comment (san7890) [inline code/modules/antagonists/heretic/items/unfathomable_curio.dm:67]:
```suggestion
	if(SEND_SIGNAL(src, COMSIG_ITEM_HIT_REACT, owner, hitby, attack_text, 0, damage, attack_type) & COMPONENT_HIT_REACTION_BLOCK && (owner.belt == src))
```

ambiguity is the mind killer

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/heretic/items/unfathomable_curio.dm [hunk 4]
```diff
@@ -44,9 +44,29 @@
...
+	atom/movable/hitby,
+	damage = 0,
+	attack_text = "the attack",
+	attack_type = MELEE_ATTACK,
+	armour_penetration = 0,
+	damage_type = BRUTE,
+)
+	SIGNAL_HANDLER
+
+	if(SEND_SIGNAL(src, COMSIG_ITEM_HIT_REACT, owner, hitby, attack_text, 0, damage, attack_type) & COMPONENT_HIT_REACTION_BLOCK && (owner.belt == src))
```
Comment (Jacquerel) [inline code/modules/antagonists/heretic/items/unfathomable_curio.dm:67]: I can see we're already doing this in at least one other place, but we should restrict signals to one source and not just fire them off like proc calls.
If the base proc `hit_reaction` isn't appropriate here, then extract this part out into a reusable proc so we can track the source of this signal more cleanly.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/heretic/items/unfathomable_curio.dm (hunk 4 above)
Comment (necromanceranne) [reply code/modules/antagonists/heretic/items/unfathomable_curio.dm:67]: Do you mean like a generic proc that sends the signal? Sorta like how health scanners do it?

Comment (LemonInTheDark) [reply code/modules/antagonists/heretic/items/unfathomable_curio.dm:67]: why are we reusing this signal at all? isn't this is what hit_reaction is like, FOR?
@Ghommie you have blame for the first case of this, what is going on?

Comment (Ghommie) [reply code/modules/antagonists/heretic/items/unfathomable_curio.dm:67]: I don't have any idea what my fault is in this. I can agree having the signals should generalize it (and perhaps have a linter test that wards us against multiple instances of the same signals being sent in the code)

**Pull request merged**